src/index.esm.js

```diff
@@ -4,3 +4,20 @@ export * from './elements';
 export * from './platform';
 export * from './plugins';
 export * from './scales';
+
+export * as controllers from './controllers';
+export * as elements from './elements';
+export * as plugins from './plugins';
+export * as scales from './scales';
+
+import * as controllers from './controllers';
+import * as elements from './elements';
+import * as plugins from './plugins';
+import * as scales from './scales';
+
+export const registerables = [
```

Comment:
Another last minute idea: instead of exposing this `registrables` array (and have to maintain it later), we could export a `registerCoreComponents()` method that abstracts this to the user:

```js
export function registerCoreComponents() {
  Chart.register(
    ...controllers,
    ...elements,
    // ...
  )
}
```

Then:

```js
import { Chart, registerCoreComponents } from 'chart.js';

registerCoreComponents();

new Chart();
```

I feel it would be easier to maintain if we either change our way to register things later. 

I think @kurkle already mentioned this approach some times ago.

I'm not sure what's my favourite solution (I would still keep exporting groups of components). 

What do you think?

Reply:
I didn't check if it works with tree-shaking.

Reply:
I don't have any objections to providing this method instead of the array. I would keep the group exports as well.

Reply:
I tested and the register function works with tree-shaking.

Reply:
I really don't know what the best API (array of registrables or this method). Also, not sure what should be the name of this method.